README.md

```diff
@@ -98,6 +98,152 @@ allows us to do:
 type Client = <Server as HasDual>::Dual;
 ```
 
+## Why is it cool?
+
+Session types are not a new concept. They are cool, because they allow for
+_compile-time_ verification of process communication. In other words, we are
+able to check statically if two communicating processes adhere to their shared
+communication protocol.
+
+But implementing session types requires a way to model that certain actions in
+the protocol have taken place. This is a complicated thing to do statically in
+most programming languages, but in Rust it is easy because of _move semantics_.
+
+Using move semantics, we ensure that "taking a step" in the protocol (sending,
+receiving, etc) cannot be repeated.
+
+## Protocol types
+
+Any session-typed communication protocol is constructed from some basic
+building blocks. This section goes through them pair-wise, showing an action
+and its dual.
+
+A session-typed channel is defined as `Chan<E, P>` where `E` is an
+_environment_ and `P` is a protocol. The environment `E` is always `()` for
+newly created channels (ie it is empty).
+
+### `Eps`
+
+This is the final step of any terminating protocol. The simplest example is:
+
+    type Close = Eps;
+
+Any channel whose type is `Chan<E, Eps>` implements the `close()` function that
+closes the connection.
+
```

Comment:
Perhaps you should mention that the Dual of an `Eps` is also `Eps`.

Reply:
Sure.